### ISSUE TITLE
内存溢出问题，我租用的云端只有32g内存第一阶段最大需要45g的内存，怎么办

### ISSUE THREAD

**Issue by ysb003** [ali-vilab/ACE_plus#114]
python run_train.py  --cfg train_config/ace_plus_lora.yaml

scepter [INFO] 2025-04-19 13:35:56,941 [File: import_utils.py Function: <module> at line 92]  PyTorch version 2.6.0 Found.
/root/miniconda3/lib/python3.11/site-packages/scepter/modules/solver/base_solver.py:189: UserWarning: No module named 'pytorch_lightning'
  warnings.warn(f'{e}')
[Info]: Loading config from [train_config/ace_plus_lora.yaml] as yaml file.
[Info]: And running with seed 1999!
scepter.ast [INFO] 2025-04-19 13:36:05,302 [File: ast_utils.py Function: load_index at line 481]  Building ast index from scanning every file!
scepter.ast [INFO] 2025-04-19 13:36:05,303 [File: ast_utils.py Function: get_files_scan_results at line 445]  AST-Scanning the path "/root/miniconda3/lib/python3.11/site-packages/scepter" with the following sub folders ['modules', 'studio', 'tools', 'workflow']
scepter.ast [INFO] 2025-04-19 13:36:06,248 [File: ast_utils.py Function: get_files_scan_results at line 467]  Scanning done! A number of 262 components indexed or updated! Time consumed 0.9458446502685547s
scepter [WARNING] 2025-04-19 13:36:06,248 [File: import_utils.py Function: import_module at line 325]  ('SOLVERS', 'FormalACEPlusSolver') not found in ast index file
/root/miniconda3/lib/python3.11/site-packages/scepter/modules/utils/distribute.py:337: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  storage = torch.ByteStorage.from_buffer(buffer)
scepter [INFO] 2025-04-19 13:36:06,652 [File: base_solver.py Function: __init__ at line 324]  Parse work dir ./examples/exp_example/20250419133605's prefix is /gemini/code/ACE_plus
scepter [INFO] 2025-04-19 13:36:06,652 [File: diffusion_solver.py Function: __init__ at line 196]  Use fsdp as the backend of ddp.
scepter [INFO] 2025-04-19 13:36:06,659 [File: logger.py Function: init_logger at line 85]  Running task with log file: /gemini/code/ACE_plus/./examples/exp_example/20250419133605/std_log.txt
scepter [WARNING] 2025-04-19 13:36:07,135 [File: import_utils.py Function: import_module at line 325]  ('DATASETS', 'ACEPlusDataset') not found in ast index file
scepter [INFO] 2025-04-19 13:36:07,136 [File: ace_plus_dataset.py Function: read_data_list at line 152]  portrait has 1 samples.
scepter [INFO] 2025-04-19 13:36:07,137 [File: registry.py Function: __init__ at line 185]  Built dataloader with len 9223372036854775807
scepter [WARNING] 2025-04-19 13:36:07,137 [File: import_utils.py Function: import_module at line 325]  ('DATASETS', 'ACEPlusDataset') not found in ast index file
scepter [INFO] 2025-04-19 13:36:07,164 [File: ace_plus_dataset.py Function: read_data_list at line 152]  portrait has 1 samples.
scepter [INFO] 2025-04-19 13:36:07,164 [File: registry.py Function: __init__ at line 185]  Built dataloader with len 1
Killed

这个是载入内存处理的时候都问题咋办。我只有32g内存，目前换不了云因为我在这个平台已经充了不少钱了

**Comment by ysb003** [comment]
对了这个可恶的平台不支持Swap，也没办法挂载其他有硬盘
